crates/autopilot/src/domain/competition/winner_selection/mod.rs

```diff
@@ -18,21 +55,39 @@ pub mod max_score;
 /// The functions assume the `Arbitrator` is the only one
 /// changing the ordering or the `participants.
 pub trait Arbitrator: Send + Sync + 'static {
+    /// Runs the entire auction mechanism on the passed in solutions.
+    fn arbitrate(&self, participants: Vec<Participant<Unranked>>, auction: &Auction) -> Ranking {
+        let partitioned = self.partition_unfair_solutions(participants, auction);
+        let filtered = partitioned
+            .filtered
+            .into_iter()
+            .map(|participant| participant.rank(Ranked::Filtered))
+            .collect();
+
+        let mut ranked = self.mark_winners(partitioned.kept);
+        ranked.sort_by_key(|participant| {
```

Comment:
This could use a unit test I think.

Reply:
This currently requires a ton of boiler plate code. Either we pull in a ton of dependencies to create `Participant<Unranked>` in the test or alternatively we introduce another type `SortingKey` and make the sorting function generic over that.
The [PR](https://github.com/cowprotocol/services/pull/3429) that originally introduced the sorting (note that the code was only moved in this PR) had a playground link to convince the reviewer that it does what it's supposed to. This sorting is now also used and verified in a unit test so I'd call this a "test by induction". 😅 
I'm open to hear other suggestions for making this more testable if you have any.